dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilder.java

```diff
@@ -58,25 +58,29 @@ public ClassPathBuilder(DependencyGraphBuilder dependencyGraphBuilder) {
    * closest' strategy follows Maven's dependency mediation.
    *
    * @param artifacts Maven artifacts to check. They are treated as the root of the dependency tree.
-   * @throws RepositoryException when there is a problem retrieving jar files
    */
-  public ClassPathResult resolve(List<Artifact> artifacts) throws RepositoryException {
+  public ClassPathResult resolve(List<Artifact> artifacts) {
 
     LinkedListMultimap<Path, DependencyPath> multimap = LinkedListMultimap.create();
     if (artifacts.isEmpty()) {
       return new ClassPathResult(multimap, ImmutableList.of());
     }
     // dependencyGraph holds multiple versions for one artifact key (groupId:artifactId)
     DependencyGraphResult result =
-        dependencyGraphBuilder.getStaticLinkageCheckDependencyGraph(artifacts);
+        dependencyGraphBuilder.buildLinkageCheckDependencyGraph(artifacts);
     List<DependencyPath> dependencyPaths = result.getDependencyGraph().list();
 
     // To remove duplicates on (groupId:artifactId) for dependency mediation
     Map<String, String> keyToFirstArtifactVersion = Maps.newHashMap();
 
     for (DependencyPath dependencyPath : dependencyPaths) {
       Artifact artifact = dependencyPath.getLeaf();
-      Path jarAbsolutePath = artifact.getFile().toPath().toAbsolutePath();
+      File file = artifact.getFile();
+      if (file == null) {
```

Comment:
not sure, but what if a file is not found for foo:bar:21.6 but we already have foo:bar:24.8 in the graph? Do we still record an unresolved artifact problem? Does this method still work?

Reply:
Yes, that unavailable artifacts are recorded in `result.getArtifactProblems()`.

Created the followup. https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1215